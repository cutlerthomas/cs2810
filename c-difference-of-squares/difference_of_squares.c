
#include <stdio.h>
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number) {
    unsigned int sum = 0;
    unsigned int i = 0;
    for ( i = 0; i < number + 1; i++) {
	unsigned int sqr = i*i;
        sum = sum + sqr;
    }
    return sum;
}

unsigned int square_of_sum(unsigned int number) {
    unsigned int sumsq = 0;
    unsigned int i = 0;
    for ( i = 0; i < number + 1; i++) {
        sumsq = sumsq + i;
    }
    sumsq = sumsq*sumsq;
    return sumsq;
}

unsigned int difference_of_squares(unsigned int number) {
    unsigned int diff = 0;
    unsigned int sumsqr = sum_of_squares(number);
    unsigned int sqrsum = square_of_sum(number);
    if (sumsqr <= sqrsum) {
    	diff = sqrsum - sumsqr;
}   else {
	diff = sumsqr - sqrsum;
}	
    return diff;
}
