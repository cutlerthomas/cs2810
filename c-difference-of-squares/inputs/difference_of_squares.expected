Testing square_of_sum with input 1
Expected result was 1, actual result was 1

Testing square_of_sum with input 5
Expected result was 225, actual result was 225

Testing square_of_sum with input 100
Expected result was 25502500, actual result was 25502500

Testing sum_of_squares with input 1
Expected result was 1, actual result was 1

Testing sum_of_squares with input 5
Expected result was 55, actual result was 55

Testing sum_of_squares with input 100
Expected result was 338350, actual result was 338350

Testing difference_of_squares with input 1
Expected result was 0, actual result was 0

Testing difference_of_squares with input 5
Expected result was 170, actual result was 170

Testing difference_of_squares with input 100
Expected result was 25164150, actual result was 25164150
